sdks/uniswapx-sdk/src/utils/OrderQuoter.ts

```diff
@@ -267,20 +278,39 @@ export class UniswapXOrderQuoter
   }
 
   /// Get the results of a multicall for a given function
+  /// Each order with a blockOverride is multicalled separately
   private async getMulticallResults(
     functionName: string,
     orders: SignedOrder[]
   ): Promise<MulticallResult[]> {
-    const calls = orders.map((order) => {
-      return [order.order.serialize(), order.signature];
+    const ordersWithBlockOverrides = orders.filter((order) => order.order.blockOverrides);
```

Comment:
could group by ones with the same blockOverrides

Reply:
that's a nice optimization, I'll do in follow up to unblock us